kattelmod/updater.py

```diff
@@ -64,6 +64,11 @@ async def _run(self) -> None:
                     logger.warn("Update task is struggling: updates take "
                                 "%g seconds but repeat every %g seconds" %
                                 (update_time, self.period))
+                    # asyncio.sleep behaviour differs between Python versions
+                    # when the sleep time is negative. From 3.7 onwards it
+                    # uses the same fast path as a negative sleep. To make
```

Comment:
What is a fast path and what does negative sleep do usually?

Reply:
Take a look at the source for `asyncio.tasks.sleep`. https://github.com/python/cpython/pull/4908 is the change that broke the tests.

Reply:
Thanks, very helpful! So if delay <= 0 it just blinks and carries on :-) Would this starve the event loop?

Reply:
Timer-based work still gets a chance to run, it just has to wait for an extra iteration or two.